packages/trader/src/AppV2/Components/TradeParameters/TakeProfit/take-profit.tsx

```diff
@@ -10,9 +10,30 @@ import CarouselHeader from 'AppV2/Components/Carousel/carousel-header';
 import TakeProfitAndStopLossInput from '../RiskManagement/take-profit-and-stop-loss-input';
 import TradeParamDefinition from 'AppV2/Components/TradeParamDefinition';
 import { TTradeParametersProps } from '../trade-parameters';
+import useTradeParamError from 'AppV2/Hooks/useTradeParamError';
+import { getDisplayedContractTypes } from 'AppV2/Utils/trade-types-utils';
 
 const TakeProfit = observer(({ is_minimized }: TTradeParametersProps) => {
-    const { currency, has_open_accu_contract, has_take_profit, is_market_closed, take_profit } = useTraderStore();
+    const {
+        currency,
+        contract_type,
+        has_open_accu_contract,
+        has_take_profit,
+        is_market_closed,
+        take_profit,
+        trade_types,
+        trade_type_tab,
+        proposal_info,
+        validation_errors,
+    } = useTraderStore();
+    const contract_types = getDisplayedContractTypes(trade_types, contract_type, trade_type_tab);
+
+    const { is_error_matching_trade_param: has_error } = useTradeParamError({
```

Comment:
For scenarios, when we haven't change the TP or SL, but already have an error for their fields (e.g. in case if Stake was changed).